package/rbd/rancher-rbd

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
+
+if [ -e "$(dirname $0)/common.sh" ]; then
+    source $(dirname $0)/common.sh
+elif [ -e "$(dirname $0)/../common/common.sh" ]; then
+    source $(dirname $0)/../common/common.sh
+fi
+DEVICE_TIMEOUT=10
+RBD_LOCK="has_mapped"
+
+init()
+{
+    print_success
+}
+
+create()
+{
+    # The OPTS variable will be populated from the input JSON as a map of key/values
+    if [ -z "${OPTS['name']}"  ]; then
+        print_error "name is required"
+    fi
+
+    local name=${OPTS['name']}
+    local size=${OPTS['size']:-"1G"}
+    local pool=${OPTS['pool']:-"rbd"}
+    local image_feature=${OPTS['image_feature']:-"layering"}
+    local OUT
+
+    OUT=$(rbd create ${pool}/${name} --size ${size} --image-feature ${image_feature})
+    if [ $? -ne 0  ]; then
+        print_error "${OUT}"
+    fi
+
+    print_options created true name ${name}
+}
+
+delete()
+{
+    # The OPTS variable will be populated from the input JSON as a map of key/values
+    if [ -z "${OPTS['name']}" ]; then
+        print_error "name is required"
+    fi
+
+    local name=${OPTS['name']}
+    local pool=${OPTS['pool']:-"rbd"}
+    local locker_id
+    local UNLOCK_OUT
+    local OUT
+
+    OUT=$(rbd info ${pool}/${name})
+    if [ $? -ne 0 ]; then
+        log_info ${pool}/${name} "Device does not exist:"
```

Comment:
Interesting note:
`log_info` prints to stderr
`print_success` prints to stdout

This is because the framework relies on parsing the output of these calls. This works great for rancher, but @ibuildthecloud just pointed out that kubernetes squashes the stderr and stdout output for flexvol calls.

So, this may fail if we ever used this in kubernetes, but our other drivers do the same thing and its not realistic to expect any of them to work in kubernetes out-of-box. So, you don't need to change anything, I just wanted to let you know.